### PR TITLE
Working towards clean-up comments /**/ in cs

### PR DESCRIPTION
**Pull request by Escoto**
Cleaning up /**/ style comments - Following https://docs.microsoft.com/en-us/dotnet/csharp/programming-guide/xmldoc/recommended-tags-for-documentation-comments

-AssemblyBuilder.cs
-AssemblyBuilderData.cs

#20330

### CODE REVIEW

File: src/System.Private.CoreLib/src/System/Reflection/Emit/AssemblyBuilder.cs [hunk 1]
```diff
@@ -277,7 +280,10 @@ private void InitManifestModule()
         /// to have a strong name and a hash will be computed when the assembly
         /// is saved.
         /// </summary>
-        [System.Security.DynamicSecurityMethod] // Methods containing StackCrawlMark local var has to be marked DynamicSecurityMethod
+        /// <remarks>
+        /// Methods containing StackCrawlMark local var has to be marked DynamicSecurityMethod.
```
Comment (jkotas) [inline src/System.Private.CoreLib/src/System/Reflection/Emit/AssemblyBuilder.cs]: This comment does not belong to the public description of the method contract. It belongs next to the `DynamicSecurityMethod` attribute like where it was.

File: src/System.Private.CoreLib/src/System/Reflection/Emit/AssemblyBuilder.cs [hunk 2]
```diff
@@ -136,9 +135,13 @@ public override string ImageRuntimeVersion
...
     }
-
-    // AssemblyBuilder class.
-    // deliberately not [serializable]
+    
+    /// <summary>
+    /// AssemblyBuilder class.
+    /// </summary>
+    /// <remarks>
+    /// Deliberately not [serializable]
```
Comment (jkotas) [inline src/System.Private.CoreLib/src/System/Reflection/Emit/AssemblyBuilder.cs]: This comment is not useful. You can delete it.

File: src/System.Private.CoreLib/src/System/Reflection/Emit/AssemblyBuilder.cs (hunk 1 above)
Comment (Escoto) [reply src/System.Private.CoreLib/src/System/Reflection/Emit/AssemblyBuilder.cs]: oh ok, let me fix that.

File: src/System.Private.CoreLib/src/System/Reflection/Emit/AssemblyBuilder.cs [hunk 3]
```diff
@@ -2,25 +2,24 @@
...
-//  2.  The "external" object is merely an wrapper of the "internal" object and all operations on it
-//      are directed to the internal object. This is the one you get by calling DefineDynamicAssembly
-//      on AppDomain and the one you can always perform RefEmit operations on. You can get other "external"
-//      objects from the "external" AssemblyBuilder, ModuleBuilder, TypeBuilder, MethodBuilder, etc. Note
-//      that VM doesn't know about this object. So every time we call into the VM we need to pass in the
-//      "internal" object.
-//
-// "internal" and "external" ModuleBuilders are similar
-//*************************************************************************************************************
+/// <summary>
```
Comment (Escoto) [inline src/System.Private.CoreLib/src/System/Reflection/Emit/AssemblyBuilder.cs]: Is this also supposed to be hidden from the public description?

Comment (jkotas) [reply src/System.Private.CoreLib/src/System/Reflection/Emit/AssemblyBuilder.cs]: Yes, I think so.

File: src/System.Private.CoreLib/src/System/Reflection/Emit/AssemblyBuilder.cs [hunk 4]
```diff
@@ -2,7 +2,6 @@
 // The .NET Foundation licenses this file to you under the MIT license.
 // See the LICENSE file in the project root for more information.
 
-
 //*************************************************************************************************************
```
Comment (jkotas) [inline src/System.Private.CoreLib/src/System/Reflection/Emit/AssemblyBuilder.cs]: If you would like to, it is fine to change the line that is full of stars `//**************************` to just `//`. It is the prevalent style that we use.

**Pull request merged**